docs/spec/openapi.yml

```diff
@@ -2156,6 +2156,219 @@ paths:
+      requestBody:
+        description: The Split-Horizon DNS Config instance object to be created.
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SplitHorizonDNSConfig'
+    get:
+      tags:
+        - SplitHorizonDNSConfig
+      summary: List Split-Horizon DNS Config instances.
+      description: This API lists all Split-Horizon DNS Config instances in a given namespace.
+      operationId: listSplitHorizonDNSConfigs
+      parameters:
+        - name: namespace
+          in: path
+          description: Namespace of the Split-Horizon DNS Config instance.
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SplitHorizonDNSConfigList'
+        '400':
+          description: Unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        '500':
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  '/namespaces/{namespace}/engine-features/split-horizon-dns-config/{name}':
+    x-everest-resource-name: engine-features/split-horizon-dns-configs
+    get:
+      tags:
+        - SplitHorizonDNSConfig
+      summary: Get Split-Horizon DNS Config instance.
+      description: This API gets the Split-Horizon DNS Config instance specified by the `name`.
+      operationId: getSplitHorizonDNSConfig
+      parameters:
+        - name: name
+          in: path
+          description: Name of the Split-Horizon DNS Config instance.
+          required: true
+          schema:
+            type: string
+        - name: namespace
+          in: path
+          description: Namespace of the Split-Horizon DNS Config instance.
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SplitHorizonDNSConfig'
+        '400':
+          description: Unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        '404':
+          description: Split-Horizon DNS Config instance not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        '500':
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    patch:
```

Comment:
PATCH assumes partial updates. In other APIs that use PATCH (update backup storages, update monitoring configs) in order to restrict what could be changed we use specific `params` types (`MonitoringInstanceUpdateParams` `UpdateBackupStorageParams`) in the request body but not the whole object. 

So in general, in the current API we have either: 
- `PUT + the whole CR`
- `PATCH + the 'params' object`

But now we want to use `PATCH + the whole CR` and I'm wondering if there is a real need for this inconsistency.

Reply:
Implemented PATCH + UpdateParams structure